src/utils/providers/ThemeProvider.tsx

```diff
@@ -0,0 +1,33 @@
+import { FC, useEffect, createContext } from "react"
+
+import { useLocalStorage, IThemeProviderProps, ThemeContextType } from "../../components/index"
+
+
+export const ThemeContext = createContext<ThemeContextType | undefined>(undefined)
+
+export const ThemeProvider: FC<IThemeProviderProps> = ({ children }) => {
+    const [theme, setTheme] = useLocalStorage("Theme", getCurrentTheme())
+
+    useEffect(() => {
```

Comment:
You could avoid this hook usage just by creating a function like 
`
function handleThemeChange(theme: TTheme) {
   setTheme(theme);
    if (theme === "Dark"){
            document.body.classList.add('dark')
     } else {
            document.body.classList.remove('dark')
     }

`
and pass it as a context value